services/api-server/src/simcore_service_api_server/utils/http_calls_capture_processing.py

```diff
@@ -52,13 +52,14 @@ def check_compatibility(cls, values):
         anyOf = values.get("anyOf")
         allOf = values.get("allOf")
         oneOf = values.get("oneOf")
+        if type_ is None and oneOf is None and anyOf is None and allOf is None:
+            type_ = "str"  # this default is introduced because we have started using json query params in the webserver
+            values["type_"] = type_
         if type_ != "str":
```

Comment:
```python

if not any([type_, oneOf, anyOf, allOf]):
    type_ = values["type_"] = "str"

# Validate that 'pattern' and 'format_' are None for types other than "str"
if type_ != "str" and any([pattern, format_]):
    raise ValueError(f"Type {type_} requires both pattern and format to be None")
```

Reply:
Thanks, this is definitely cleaner. I will adapt.